pkg/core/pipeline/targets.go

```diff
@@ -38,41 +36,27 @@ func (p *Pipeline) RunTargets() error {
 		target := p.Targets[id]
 		target.Config = p.Config.Targets[id]
 
-		rpt := p.Report.Targets[id]
+		report := p.Report.Targets[id]
 		// Update report name as the target configuration might has been updated (templated values)
-		rpt.Name = target.Config.Name
+		report.Name = target.Config.Name
 
 		err = target.Run(p.Sources[target.Config.SourceID].Output, &p.Options.Target)
-
-		rpt.Result = target.Result
-
 		if err != nil {
-			logrus.Errorf("Something went wrong in target \"%v\" :\n", id)
-			logrus.Errorf("%v\n\n", err)
-
-			isResultIsFailed = true
-
-			p.Targets[id] = target
-			p.Report.Targets[id] = rpt
-			i++
-			continue
-		}
-
-		if strings.Compare(target.Result, result.ATTENTION) == 0 {
-			isResultIsChanged = true
+			p.Report.Result = result.FAILURE
```

Comment:
I didn't test it but I think, we also need to save the target state using 

```suggestion
			p.Report.Result = result.FAILURE
			p.Targets[id] = target
```

Reply:
`p.Targets[id]` already contains `target` here (AFAICT), as it's where it comes from (cf line 36), I don't think it should be reset. WDYT?

Reply:
These are 2 different objects, because of https://github.com/lemeurherve/updatecli/blob/is632-report-failure/pkg/core/pipeline/main.go#L26 which is not a pointer.

It means that https://github.com/updatecli/updatecli/pull/638/files#diff-de07e8be2bb72a5be95eaabc19216134364a70a5bf9a98c2903c41adb0477098R36 creates an entire different object into the variable `target` by copy.

Since this object is mutated later (https://github.com/updatecli/updatecli/pull/638/files#diff-de07e8be2bb72a5be95eaabc19216134364a70a5bf9a98c2903c41adb0477098R37, eventually during https://github.com/updatecli/updatecli/pull/638/files#diff-de07e8be2bb72a5be95eaabc19216134364a70a5bf9a98c2903c41adb0477098R43, etc.)
then you need to ensure that the pipeline is updated with the object at the end (otherwise the pullrequest would be executed with the original object, not the updated one).


Pleasures of low level languages :)

Reply:
Not sure if returning an updated but also failed target is really useful as we're returning an error. Added neverless.